data/model/repository.py

```diff
@@ -739,17 +739,18 @@ def force_cache_repo_size(repo_id: int):
     except Manifest.DoesNotExist:
         size = 0
 
-    repo = get_repository_size(repo_id)
-
     if size is None:
         size = 0
 
-    if repo is not None:
```

Comment:
I dont think this is going to fix what your trying to fix. If this is a race condition as you explain in the ticket this PR is linked too this will not fix that as 2 executions reaching the get_repo_size function at the same time will still error.